boards/common/stm32-stlink/Makefile.include

```diff
@@ -0,0 +1,14 @@
+# configure the serial terminal
+PORT_LINUX ?= /dev/ttyACM0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
+include $(RIOTMAKE)/tools/serial.inc.mk
```

Comment:
I think it is not very clean to put the common serial configuration into a common board folder that is referring to a programmer...

I think I have an alternative idea, will PR something later tonight.

Reply:
forgot to mention: examples where this might get confusion are for example the green discovery boards, which have an on-board stlink, but don't supply any serial adapter...

Reply:
ok, I was not aware of that.

> I think I have an alternative idea, will PR something later tonight.

Describe what you have in mind and I'll update this PR

Reply:
Alright, so how about #8484 for this part.